platforms/common/include/openmm/common/ComputeContext.h

```diff
@@ -387,6 +387,12 @@ class OPENMM_EXPORT_COMMON ComputeContext {
     const std::vector<int>& getAtomIndex() const {
         return atomIndex;
     }
+    /**
+     * Set the host-side vector which contains the index of each atom.
+     */
+    void setAtomIndex(std::vector<int>& index){
+        atomIndex = index;
+    }
```

Comment:
The atom indices get stored on both host and device.  We need to update them in both places.  We also need to inform any ReorderListeners that the order has changed.  You can do those like this:

```c++
getAtomIndexArray().upload(atomIndex);
for (auto listener : reorderListeners)
    listener->execute();
```

And update the comment to remove "host-side".